usr/src/cmd/zdb/zdb.c

```diff
@@ -1578,8 +1583,54 @@ dump_deadlist(dsl_deadlist_t *dl)
 static avl_tree_t idx_tree;
 static avl_tree_t domain_tree;
 static boolean_t fuid_table_loaded;
-static boolean_t sa_loaded;
-sa_attr_type_t *sa_attr_table;
+static objset_t *sa_os = NULL;
```

Comment:
what's the point of `sa_os`?  For assertion purposes (that we don't open an objset when one is already open), you could use sa_attr_table.